tooling/metamodel-generator/src/main/java/org/hibernate/processor/HibernateProcessor.java

```diff
@@ -362,59 +365,70 @@ private void processClasses(RoundEnvironment roundEnvironment) {
 		}
 
 		for ( Element element : roundEnvironment.getRootElements() ) {
-			try {
-				if ( !included( element )
-						|| hasAnnotation( element, Constants.EXCLUDE )
-						|| hasPackageAnnotation( element, Constants.EXCLUDE ) ) {
-					// skip it completely
-				}
-				else if ( isEntityOrEmbeddable( element ) ) {
-					context.logMessage( Diagnostic.Kind.OTHER, "Processing annotated entity class '" + element + "'" );
-					handleRootElementAnnotationMirrors( element );
-				}
-				else if ( hasAuxiliaryAnnotations( element ) ) {
-					context.logMessage( Diagnostic.Kind.OTHER, "Processing annotated class '" + element + "'" );
-					handleRootElementAuxiliaryAnnotationMirrors( element );
+			processElement( element );
+		}
+	}
+
+	private void processElement(Element element) {
+		try {
+			if ( !included( element )
+					|| hasAnnotation( element, Constants.EXCLUDE )
+					|| hasPackageAnnotation( element, Constants.EXCLUDE ) ) {
+				// skip it completely
+			}
+			else if ( isEntityOrEmbeddable( element ) && !element.getModifiers().contains( Modifier.PRIVATE )) {
```

Comment:
I think we could just say that the processor completely skips `private` inner types. WDYT of this @cigaly:

```suggestion
			if ( !included( element )
					|| hasAnnotation( element, Constants.EXCLUDE )
					|| hasPackageAnnotation( element, Constants.EXCLUDE ) 
					|| element.getModifiers().contains( Modifier.PRIVATE ) ) {
				// skip it completely
			}
			else if ( isEntityOrEmbeddable( element ) ) {
```

Reply:
Agree. And for maybe will be good idea to keep access modifier for non private classes?

Reply:
> And for maybe will be good idea to keep access modifier for non private classes?

You mean generate the metamodel with the same visibility? Yeah, that makes sense I suppose...

Reply:
One more thing ... in PR [#9220](https://github.com/hibernate/hibernate-orm/pull/9220) id class is generated  as inner class of parent meta model class. Here meta model class for each inner class is generated as separate class. I guess that now is good time to decide if this should stay that way or will be better idea to generate them as inner classes as well.

Reply:
I think I would prefer it to be an inner class.

Reply:
Agree, I will try to refactor this to make things that way some time next week.